src/librbd/ImageCtx.h

```diff
@@ -225,6 +226,8 @@ namespace librbd {
     exclusive_lock::Policy *exclusive_lock_policy = nullptr;
     journal::Policy *journal_policy = nullptr;
 
+    crypto::CryptoInterface *crypto = nullptr;
```

Comment:
Nit: totally fine for now, but we will eventually want to move the core crypto dispatch logic to a librbd plug-in like the parent object cache.

Reply:
Makes sense. I will re-factor this PR as a plugin